packages/block-library/src/template-part/edit/index.js

```diff
@@ -72,7 +72,12 @@ export default function TemplatePartEdit( {
 	const isPlaceholder = ! slug;
 	const isEntityAvailable = ! isPlaceholder && ! isMissing;
 
-	if ( ! isPlaceholder && isMissing ) {
+	// We don't want to render a missing state if we have any inner blocks.
+	// A new template part is automatically created if we have any inner blocks but no entity.
+	if (
+		innerBlocks.length === 0 &&
+		( ( slug && ! theme ) || ( slug && isMissing ) )
```

Comment:
> A new template part is automatically created if we have any inner blocks but no entity.

Trying to wrap my head around why this happens 🤔. The rest of the conditional makes sense to me.

Reply:
Did some debugging and still a bit stumped by where the automatic template part creation happens. Would you mind elaborating more on this?

Reply:
Don't worry, it took me a while to figure out as well :D `TemplatePartPlaceholder` component takes care of the creation.

https://github.com/WordPress/gutenberg/blob/f31c35c5756091392f64c203c1e0a57c6badab8e/packages/block-library/src/template-part/edit/placeholder/index.js#L39-L52

Reply:
The changes look good 👍